src/TransitionReplace/index.jsx

```diff
@@ -0,0 +1,176 @@
+  // both the entering and leaving children
+  //
+  // We set the container height (for animation) in this order:
+  //
+  // 1. onChildExit         (explicitly set the height to match the current current)
+  // 2. onChildEntering     (set the height to match the new content)
+  // 3. onChildExited       (reset the height to null)
+
+  onChildEnter(htmlNode) {
+    if (this.props.onChildEnter) this.props.onChildEnter(htmlNode);
+  }
+
+  onChildEntering(htmlNode) {
+    this.setState({ height: htmlNode.offsetHeight });
+    if (this.props.onChildEntering) this.props.onChildEntering(htmlNode);
+  }
+
+  onChildEntered(htmlNode) {
+    this.setState({ height: null });
+    if (this.props.onChildEntered) this.props.onChildEntered(htmlNode);
+  }
+
+  onChildExit(htmlNode) {
+    this.setState({ height: htmlNode.offsetHeight });
+    if (this.props.onChildExit) this.props.onChildExit(htmlNode);
+  }
+
+  onChildExiting(htmlNode) {
+    if (this.props.onChildExiting) this.props.onChildExiting(htmlNode);
+  }
+
+  onChildExited(htmlNode) {
+    this.setState({ height: null });
+    if (this.props.onChildExited) this.props.onChildExited(htmlNode);
+  }
+
+  renderChildTransition(child) {
+    if (!child.key && process.env.NODE_ENV === 'development') {
+      // eslint-disable-next-line no-console
+      console.warn("TransitionReplace: A child is missing a 'key' prop. Keys are required for any child of this component.");
+    }
+    // Makes the exiting and entering children occupy the same space
+    // SCSS handles the crossfade so it can be easily overridden
+    const commonChildStyles = {
+      // Prevent margin-collapsing which throws off height calculations
+      padding: '.1px 0',
+    };
+
+    const transitionStyles = {
+      entering: {},
+      entered: {},
+      exiting: {
+        position: 'absolute',
+        top: 0,
+        left: 0,
+        right: 0,
+        pointerEvents: 'none',
+      },
+      exited: {},
+    };
+
+    return (
+      <CSSTransition
+        timeout={{
+          enter: this.props.enterDuration,
+          exit: this.props.exitDuration,
+        }}
+        unmountOnExit
+        mountOnEnter
+        onEnter={this.onChildEnter}
+        onEntering={this.onChildEntering}
+        onEntered={this.onChildEntered}
+        onExit={this.onChildExit}
+        onExiting={this.onChildExiting}
+        onExited={this.onChildExited}
+        classNames={this.props.transitionClassNames}
+      >
+        {state => (
+          <div
+            style={{
+              ...commonChildStyles,
+              ...transitionStyles[state],
+              ...this.props.transitionStyles[state],
+            }}
+          >
+            {child}
+          </div>
+        )}
+      </CSSTransition>
+    );
+  }
+
+  render() {
+    return (
+      <TransitionGroup
+        className={classNames('pgn-transition-replace-group', this.props.className)}
+        style={{
+          position: 'relative',
```

Comment:
Why not just use the util class ```position-relative```?

Reply:
Per this feedback, I've swapped using style to prefer bootstrap utility classes where possible. I have no particular preference on one over the other, maybe we can discuss in person pros/cons of inline styles vs utility classes in varying situations.